src/app/my-messages/deletable-notification/deletable-notification.component.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { async, ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+
+import { DeletableNotificationComponent } from './deletable-notification.component';
+import { MatDialog } from '@angular/material';
+import { DeleteConfirmationDialogComponent } from '../../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { NotificationType } from '../notification-type.enum';
+import { MessagesService } from '../messages.service';
+import { AppMaterialModule } from '../../app-material.module';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatMomentDateModule } from '@angular/material-moment-adapter';
+import { ReactiveFormsModule } from '@angular/forms';
+import { RouterTestingModule } from '@angular/router/testing';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+import { InfoDialogComponent } from '../../shared/info-dialog/info-dialog.component';
+
+const notification =  {
+  type: NotificationType.USER_WELCOME_NOTIFICATION,
+  id: '997f8c9e-4655-47f7-8cf0-b6021b25405c',
+  creationDatetime: new Date()
+};
+
+describe('DeletableNotificationComponent', () => {
```

Comment:
If `DeletableNotificationComponent` is turned into the abstract class `DeletableNotificationComponentTrait` it will be impossible to instantiate it anymore. But one of the classes using the `DeletableNotificationComponentTrait` can still be instantiated and used for tests.